beacon-chain/sync/initial-sync/round_robin.go

```diff
@@ -73,6 +73,11 @@ func (s *InitialSync) roundRobinSync(genesis time.Time) error {
 			// Handle block large block ranges of skipped slots.
 			start += count * uint64(lastEmptyRequests*len(peers))
 
+			// Short circuit start far exceeding the highest finalized epoch in some infinite loop.
+			if start > helpers.StartSlot(highestFinalizedEpoch()+1) {
```

Comment:
Any way this case can be covered in a test? Seems very important

Reply:
+1

Reply:
This scenario is not very well understood. Lets file and issue and capture the conditions when it happens in the next testnet.